### PR TITLE
Feature/filters product list

### PR DESCRIPTION
**Pull request by Pawelooo**
I created a search for products only on active ones. I don't know if it's not enough to search for not active products ?
At Orderingfilter, he returns Z-A products first and not the A-Z product first
A-Z product
Product with Z-A and next Product A-Z

### CODE REVIEW

File: Dshop/apps/products_catalogue/filters.py [hunk 1]
```diff
@@ -1,11 +1,41 @@
+from django.forms import TextInput
 from django_filters import rest_framework as filters
-from .models import Product
+from .models import Product, Category
 
 
 class ProductFilter(filters.FilterSet):
-    name = filters.CharFilter(field_name='name', lookup_expr='icontains')
-    is_active = filters.BooleanFilter(field_name='is_active')
+    name = filters.CharFilter(field_name='name', lookup_expr='icontains',
```
Comment (jacoor) [inline Dshop/apps/products_catalogue/filters.py:7]: Please add tests.  You can test by checking page context, like here: https://stackoverflow.com/a/50225148
If you don't know how, let me know, I'll guide you.

**Review by jacoor — CHANGES_REQUESTED** [review]

File: Dshop/apps/products_catalogue/filters.py [hunk 2]
```diff
@@ -1,11 +1,55 @@
...
+
+    def filter_availability(self, queryset, name, value):
+        value = int(value)
+        if value in (1, 3, 7, 14):
+            return queryset.filter(availability__lte=value)
+        return queryset.filter(**{name: value})
+
+    def __init__(self, *args, **kwargs):
+        super(ProductFilter, self).__init__(*args, **kwargs)
+        self.filters['name'].label = 'Nazwa produktu'
```
Comment (yanazPL) [inline Dshop/apps/products_catalogue/filters.py:47]: Not sure if those are needed. labels already defined here:
```python
field_labels={
            'price': 'Cena rosnąco',
            '-price': 'Cena malejąco',
            '-name': 'Produkt A-Z',
            'name': 'Produkt Z-A',
            'created_at': 'Najnowsze',
            '-created_at': 'Najstarsze',
        },
```

**Comment by yanazPL** [comment]
I just added the newest, the oldest product order filters.

Also availability filters have been added.

Tests of `ProductListView` left for you @Pawelooo 
I'll write some for API view.

File: Dshop/apps/products_catalogue/filters.py (hunk 2 above)
Comment (Pawelooo) [reply Dshop/apps/products_catalogue/filters.py:47]: I don't know if it's okay either, but I need the eye of an experienced programmer @jacoor

Comment (jacoor) [reply Dshop/apps/products_catalogue/filters.py:47]: Ha. I don't know :-) 
You need to experiment with it and see if it works as intended or is needed. 
That's what I would do.

**Comment by jacoor** [comment]
@Pawelooo you need to update from dev.

**Comment by jacoor** [comment]
I updated branch from dev, so pull.

File: Dshop/apps/products_catalogue/tests/test_products_filters.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+def test_filter_by_name(client, products):
+    url = reverse('product-list')
+    response = client.get(f'{url}?order_by=name')
+    products = response.context['object_list']
+    for idx, name in enumerate(range(1, 11)):
+        assert products[idx].name == f"main product ${idx}"
+
+
+
+# @pytest.mark.django_db
```
Comment (jacoor) [inline Dshop/apps/products_catalogue/tests/test_products_filters.py]: Please remove those comments

**Review by jacoor — CHANGES_REQUESTED** [review]

**Comment by jacoor** [comment]
@Pawelooo please fix tests
https://github.com/pymasterspl/Dshop/actions/runs/8046459105/job/21973713368

**Comment by Pawelooo** [comment]
@jacoor updated

**Review by jacoor — APPROVED** [review]

**Pull request merged**